src/super_gradients/training/pipelines/pipelines.py

```diff
@@ -266,31 +266,83 @@ def __init__(
         :param model_input:     Model input (i.e. images after preprocessing).
+
+        :param target_bboxes: Optional[List[np.ndarray]], ground truth bounding boxes. Can either be an np.ndarray of shape
+         (image_i_object_count, 4) when predicting a single image, or a list of length len(target_bboxes), containing such arrays.
+         When not None, will plot the predictions and the ground truth bounding boxes side by side (i.e 2 images stitched as one).
+
+        :param target_class_ids: Optional[List[np.ndarray]], ground truth target class indices. Can either be an np.ndarray of shape
+         (image_i_target_count) when predicting a single image, or a list of length len(target_bboxes), containing such arrays (default=None).
+
+        :param target_bboxes_format: Optional[str], bounding box format of target_bboxes, one of ['xyxy','xywh',
+        'yxyx' 'cxcywh' 'normalized_xyxy' 'normalized_xywh', 'normalized_yxyx', 'normalized_cxcywh']. Will raise an
+        error if not None and target_bboxes is None.
+
         :return:                Predicted Bboxes.
         """
+        target_bboxes, target_class_ids = self._check_target_args(target_bboxes, target_bboxes_format, target_class_ids)
+
         post_nms_predictions = self.post_prediction_callback(model_output, device=self.device)
+        if target_bboxes is None:
+            target_bboxes = [None for _ in range(len(model_input))]
+            target_class_ids = [None for _ in range(len(model_input))]
 
         predictions = []
-        for prediction, image in zip(post_nms_predictions, model_input):
+        for prediction, image, target_bbox, target_class_id in zip(post_nms_predictions, model_input, target_bboxes, target_class_ids):
             prediction = prediction if prediction is not None else torch.zeros((0, 6), dtype=torch.float32)
+            target_bbox = target_bbox if target_bbox is not None else np.zeros((0, 4))
+            target_class_id = target_class_id if target_class_id is not None else np.zeros((0, 1))
             prediction = prediction.detach().cpu().numpy()
+
             predictions.append(
                 DetectionPrediction(
                     bboxes=prediction[:, :4],
                     confidence=prediction[:, 4],
                     labels=prediction[:, 5],
                     bbox_format="xyxy",
+                    target_bboxes=target_bbox,
+                    target_labels=target_class_id,
+                    target_bbox_format=target_bboxes_format,
                     image_shape=image.shape,
                 )
             )
 
         return predictions
 
+    @staticmethod
+    def _check_target_args(
+        target_bboxes: Optional[Union[np.ndarray, List[np.ndarray]]] = None,
+        target_bboxes_format: Optional[str] = None,
+        target_class_ids: Optional[Union[np.ndarray, List[np.ndarray]]] = None,
+    ):
+        if (
+            (target_bboxes is None and target_bboxes_format is not None)
```

Comment:
too complex...
what is the logic? target ids are optional? 
```
if target_bboxes is None and target_bboxes_format is None and target_class_ids is None:
    return None, None, None
if target_bboxes is None or target_bboxes_format is None or target_class_ids is None:
    raise ValueError(..)
```

Reply:
it just means what the error says, I changed the statement to a more clear one in 16bd98c744f224f04057a99af2dfc48ee77acc96